src/templates/modules/FM_Template_v1.sol

```diff
@@ -0,0 +1,196 @@
+    // Libraries
+
+    using SafeERC20 for IERC20;
+
+    //==========================================================================
+    // ERC165
+
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(bytes4 interfaceId_)
+        public
+        view
+        virtual
+        override(Module_v1)
+        returns (bool)
+    {
+        return interfaceId_ == type(IFM_Template_v1).interfaceId
+            || interfaceId_ == type(IFundingManager_v1).interfaceId
+            || super.supportsInterface(interfaceId_);
+    }
+
+    //==========================================================================
+    // Constants
+
+    // Add constants here
+
+    //==========================================================================
+    // State
+
+    /// @dev    Mapping of user addresses to their deposited token amounts.
+    mapping(address user => uint amount) internal _depositedAmounts;
+
+    /// @dev    The orchestrator token.
+    IERC20 internal _orchestratorToken;
+
+    /// @dev    Storage gap for future upgrades.
+    uint[50] private __gap;
+
+    //==========================================================================
+    // Modifiers
+
+    // Add modifiers here
+
+    //==========================================================================
+    // Constructor & Init
+
+    /// @inheritdoc Module_v1
+    function init(
+        IOrchestrator_v1 orchestrator_,
+        Metadata memory metadata_,
+        bytes memory configData_
+    ) external override(Module_v1) initializer {
+        __Module_init(orchestrator_, metadata_);
+
+        // Decode module specific init data through use of configData bytes.
+        // This value is an example value used to showcase the setters/getters
+        // and internal functions/state formatting style.
+        (address orchestratorTokenAddress) = abi.decode(configData_, (address));
+
+        // Set init state.
+        _orchestratorToken = IERC20(orchestratorTokenAddress);
+    }
+
+    //==========================================================================
+    // Public (Getters)
+
+    /// @inheritdoc IFM_Template_v1
+    function getDepositedAmount(address user_)
+        external
+        view
+        virtual
+        override
+        returns (uint amount_)
+    {
+        uint amount_ = _depositedAmounts[user_];
+        return amount_;
+    }
+
+    /// @inheritdoc IFundingManager_v1
+    function token() external view override returns (IERC20) {
+        return _orchestratorToken;
+    }
+
+    //==========================================================================
+    // Public (Mutating)
+
+    /// @inheritdoc IFM_Template_v1
+    function deposit(uint amount_) external virtual override {
+        // Validate parameters.
+        if (amount_ == 0) {
+            revert Module__FM_Template_InvalidAmount();
+        }
+
+        // Emit event.
+        emit Deposited(_msgSender(), amount_);
```

Comment:
We usually want to have events at the end, unless there is a reason (i.e. events that want to reference before- and after-values). But here, I'd move it to the very end.